DistroLauncher/splash_controller.h

```diff
@@ -0,0 +1,358 @@
+            // this function ends, being whatever return path or exception thrown, the function `CloseHandle` will be
+            // invoked effectively closing the `HANDLE pipe` variable. That is only possible because Win32 HANDLE is a
+            // pointer type (typically void*). This using statement is just a syntactic sugar.
+            using defer = std::unique_ptr<std::remove_pointer<HANDLE>::type, decltype(&::CloseHandle)>;
+
+            constexpr int connectionTimeout = 5000; // ms.
+            const wchar_t* pipeName{L"\\\\.\\pipe\\Flutter_HWND_Pipe"};
+            // CreateThePipe
+            SECURITY_ATTRIBUTES pipeSecurity{sizeof(pipeSecurity), nullptr, true};
+            HANDLE pipe = CreateNamedPipe(pipeName,
+                                          PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
+                                          PIPE_TYPE_MESSAGE | PIPE_READMODE_MESSAGE | PIPE_WAIT,
+                                          PIPE_UNLIMITED_INSTANCES,
+                                          0,
+                                          0,
+                                          0,
+                                          &pipeSecurity);
+            if (pipe == INVALID_HANDLE_VALUE) {
+                std::wcerr << L"Failed to create pipe <" << pipeName << L">.\n";
+                return nullptr;
+            }
+            // One can read this as `defer(CloseHandle(pipe));`
+            defer pipeCleaner(pipe, &CloseHandle);
+            SetHandleInformation(pipe, HANDLE_FLAG_INHERIT, 1);
+
+            // Setup the wait for connection until timeout.
+            OVERLAPPED sync;
+            ZeroMemory(&sync, sizeof(sync));
+            sync.hEvent = CreateEvent(NULL, TRUE, FALSE, nullptr);
+            if (sync.hEvent == INVALID_HANDLE_VALUE || sync.hEvent == 0) {
+                std::wcerr << L"Failed to create internal event to notify splash connection\n";
+                return nullptr;
+            }
+            // `defer(CloseHandle(sync.hEvent));`
+            defer eventCleaner(sync.hEvent, &CloseHandle);
+            if (ConnectNamedPipe(pipe, &sync) == FALSE && GetLastError() != ERROR_PIPE_CONNECTED &&
+                GetLastError() != ERROR_IO_PENDING) {
+                std::wcerr << L"Failed to connect to splash process\n";
+                return nullptr;
+            }
+            // Block this thread until [connectionTimeout] or the OS notifies the named pipe had a client connected
+            // through the event [sync.hEvent].
+            if (WaitForSingleObject(sync.hEvent, connectionTimeout) != 0) {
+                std::wcerr << L"Timeout: no response from the splash process\n";
+                return nullptr;
+            }
+
+            // Client connected. Time to read.
+            HWND window = nullptr;
+            void* lpBuffer{&window};
+            DWORD bytesRead = -1;
+            BOOL readSuccess = FALSE;
+            constexpr DWORD bytesExpected = sizeof(window);
```

Comment:
Suspicious usage of ‘sizeof(A*)’
A common mistake is to compute the size of a pointer instead of its pointee.These cases may occur because of explicit cast or implicit conversion.
In this case I want the size of the pointer, but in a portable way.